test/e2e_node/image_pull_test.go

```diff
@@ -0,0 +1,241 @@
+							framework.ExpectNoError(err)
+							break
+						}
+					}
+				}
+			}
+			gomega.Expect(len(testpods)).To(gomega.BeComparableTo(len(httpdPulled)))
+
+			// as this is parallel image pulling, the waiting duration should be similar with the pulled duration.
+			// use 1.2 as a common ratio
+			for _, pulled := range httpdPulled {
+				if float32(pulled.pulledIncludeWaitingDuration/time.Millisecond)/float32(pulled.pulledDuration/time.Millisecond) > 1.2 {
+					framework.Failf("the pull duration including waiting %v should be similar with the pulled duration %v",
+						pulled.pulledIncludeWaitingDuration, pulled.pulledDuration)
+				}
+			}
+		})
+
+	})
+})
+
+var _ = SIGDescribe("Pull Image", framework.WithSerial(), nodefeature.MaxParallelImagePull, func() {
+
+	f := framework.NewDefaultFramework("serialize-pull-image-test")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+
+	ginkgo.Context("serialize image pull", func() {
+		// this is the default behavior now.
+		tempSetCurrentKubeletConfig(f, func(ctx context.Context, initialConfig *kubeletconfig.KubeletConfiguration) {
+			initialConfig.SerializeImagePulls = true
+			initialConfig.MaxParallelImagePulls = ptr.To[int32](1)
+		})
+
+		var testpods []*v1.Pod
+
+		ginkgo.BeforeEach(func(ctx context.Context) {
+			testpods = prepareAndCleanup(ctx, f)
+		})
+
+		ginkgo.AfterEach(func(ctx context.Context) {
+			ginkgo.By("cleanup pods")
+			for _, pod := range testpods {
+				deletePodSyncByName(ctx, f, pod.Name)
+			}
+		})
+
+		ginkgo.It("should be waiting more", func(ctx context.Context) {
+			for _, testpod := range testpods {
+				pod := e2epod.NewPodClient(f).Create(ctx, testpod)
+				err := e2epod.WaitForPodCondition(ctx, f.ClientSet, f.Namespace.Name, pod.Name, "Running", 30*time.Second, func(pod *v1.Pod) (bool, error) {
+					if pod.Status.Phase == v1.PodRunning {
+						return true, nil
+					}
+					return false, nil
+				})
+				framework.ExpectNoError(err)
+			}
+
+			events, err := f.ClientSet.CoreV1().Events(f.Namespace.Name).List(ctx, metav1.ListOptions{})
+			framework.ExpectNoError(err)
+			var httpdPulled []*pulledStruct
+			for _, event := range events.Items {
+				if event.Reason == kubeletevents.PulledImage {
+					for _, testpod := range testpods {
+						if event.InvolvedObject.Name == testpod.Name {
+							pulled, err := getDurationsFromPulledEventMsg(event.Message)
+							httpdPulled = append(httpdPulled, pulled)
+							framework.ExpectNoError(err)
+							break
+						}
+					}
+				}
+			}
+			gomega.Expect(len(testpods)).To(gomega.BeComparableTo(len(httpdPulled)))
+
+			// as this is serialize image pulling, the waiting duration should be almost double the duration with the pulled duration.
+			// use 1.5 as a common ratio to avoid some overlap during pod creation
+			if float32(httpdPulled[1].pulledIncludeWaitingDuration/time.Millisecond)/float32(httpdPulled[1].pulledDuration/time.Millisecond) < 1.5 &&
```

Comment:
should this be "> 1.5"? Pull time including waiting should be at least double the time without waiting, right?

Reply:
This is the failing condition. 

Either pulling A time should double pulling B time or pulling B time should double pulling A time.

If both < 1.5, it is not expected.